pandas/_libs/window/aggregations.pyx

```diff
@@ -1138,6 +1138,79 @@ def roll_quantile(const float64_t[:] values, ndarray[int64_t] start,
 
     return output
 
+def roll_rank(const float64_t[:] values, ndarray[int64_t] start,
+              ndarray[int64_t] end, int64_t minp, bint percentile) -> np.ndarray:
+    """
+    O(N log(window)) implementation using skip list
+
+    derived from roll_quantile
+    """
+    cdef:
+        Py_ssize_t i, j, s, e, N = len(values), idx
+        int rank = 0
+        int64_t nobs = 0, win
+        float64_t val
+        skiplist_t *skiplist
+        ndarray[float64_t] output
```

Comment:
Since we define output from `np.empty`, this could have contiguity specified for small perf improvement, `float64_t[::1] output`

Reply:
Does it make sense that I need to use `np.asarray` on `output` after making this change?